apps/api/src/app/events/usecases/trigger-event/message-matcher.service.ts

```diff
@@ -65,13 +61,39 @@ export class MessageMatcher {
 
         const singleRule = !children || (Array.isArray(children) && children.length === 1);
         if (singleRule) {
-          return await this.processFilter(variables, children[0], command);
+          const result = await this.processFilter(variables, children[0], command, filterProcessingDetails);
```

Comment:
I couldn't find a better way instead of passing this instance as the argument to the function and propagating it down the calls. The thing is that I couldn't define it as the instance variable, because this service is a singleton. I also cannot change the injection scope to the request, because this service is called from the job queue, so it's not in the request context.